nuget/lib/dependabot/nuget/update_checker/nupkg_fetcher.rb

```diff
@@ -43,11 +44,48 @@ def self.fetch_nupkg_buffer_from_repository(repository_details, package_id, pack
       end
 
       def self.get_nuget_v3_package_url(repository_details, package_id, package_version)
-        base_url = repository_details[:base_url].delete_suffix("/")
+        base_url = repository_details[:base_url]
+        unless base_url
+          return get_nuget_v3_package_url_from_search(repository_details, package_id,
+                                                      package_version)
+        end
+
+        base_url = base_url.delete_suffix("/")
         package_id_downcased = package_id.downcase
         "#{base_url}/#{package_id_downcased}/#{package_version}/#{package_id_downcased}.#{package_version}.nupkg"
       end
 
+      # rubocop:disable Metrics/PerceivedComplexity
+      def self.get_nuget_v3_package_url_from_search(repository_details, package_id, package_version)
+        search_url = repository_details[:search_url]
+        return nil unless search_url
+
+        # get search result
+        search_result_response = fetch_url(search_url, repository_details)
+        return nil unless search_result_response.status == 200
+
+        search_response_body = HttpResponseHelpers.remove_wrapping_zero_width_chars(search_result_response.body)
+        search_results = JSON.parse(search_response_body)
+
+        # find matching package and version
+        package_search_result = search_results&.[]("data")&.find { |d| package_id.casecmp?(d&.[]("id")) }
+        version_search_result = package_search_result&.[]("versions")&.find do |v|
+          package_version.casecmp?(v&.[]("version"))
+        end
+        registration_leaf_url = version_search_result&.[]("@id")
+
+        registration_leaf_response = fetch_url(registration_leaf_url, repository_details)
```

Comment:
I see a lot of conditional access prior to this point. Will `fetch_url` fail gently in the case that `registration_leaf_url` is `nil`?

Reply:
Good catch, added some early returns.